app/controllers/books_controller.rb

```diff
@@ -17,21 +17,25 @@ def new
   end
 
   def create
+    binding.pry
     @book = Book.new(book_params)
     @genres_collection = genres_collection
-
     @author = Author.find_or_create_by(name: author_params[:author])
-
     @book.author = @author
-
     @user = current_user
-
     @book.user = @user
 
     if @book.save
       flash[:success] = "Book added successfully!"
       redirect_to books_path
+    elsif @book.author.id == nil
+      @book.author = nil
+      flash[:warning] = "#{@author.errors.messages.values.to_s.
+                          gsub(/[^0-9a-z ]/i, '')}, " +
+                        @book.errors.full_messages.join(', ')
```

Comment:
We could, but there is only one Author error - the name. The Author Model was set up that it will just autopopulate "Name can't be blank" as opposed to "Author Name can't be blank".  The regex was just there to do a one-line standardization of the errors stylistically and it is functional.